skopt/optimizer/optimizer.py

```diff
@@ -166,6 +168,11 @@ def _check_arguments(self, base_estimator, n_random_starts, acq_optimizer):
             warnings.warn("The 'auto' option for the acq_optimizer will be "
                           "removed in 0.4.")
             acq_optimizer = "lbfgs"
+
+        if isinstance(base_estimator, (DecisionTreeRegressor,
+                                       ExtraTreesRegressor)):
```

Comment:
I think the motivation for this was to not let people use bfgs when using one of the [tree based surrogate models](https://github.com/scikit-optimize/scikit-optimize/blob/master/skopt/learning/__init__.py#L3-L5). So we should check for those (no need to check for DT).

---

The models in `skopt.learnin` are slight modifications on top of the scikit-learn ones. The ones in `learning` have hade an additional argument added `return_std`.